tests/test_package_generation.py

```diff
@@ -36,26 +36,18 @@ def get_all_files_folders(root_path: pathlib.Path) -> set[pathlib.Path]:
 
 
 def test_package_generation(
-    tmp_path: pathlib.Path,
-    generate_package: typing.Callable,
+    default_config_with: typing.Callable, generate_package: typing.Callable
```

Comment:
This is because we've disabled `COM812` due to incompatibility with the formatter but looks much nicer IMO

```suggestion
    default_config_with: typing.Callable,
    generate_package: typing.Callable,
```

Reply:
¯\_(ツ)_/¯ isn't this the job of the linter?

Reply:
Nope. Sorry. @pre-commit-ci just put this back as it was.

Reply:
No. It is because we have to turn off `COM812` because `ruff-format` warns it would clash with `ruff-lint`. In 99% of cases, it doesn't clash. The only time it does is the use case described here https://github.com/astral-sh/ruff/issues/9216. What I tend to do is uncomment that rule briefly (which is what would have happened here) run the linting and then restore it.

Reply:
>  uncomment that rule briefly

...? I'm also a bit 👎 about this level of faffing with a linter. I want set-and-forget. Isn't the whole point of the linter automation that I never have to spend time thinking about this?

Reply:
We could just leave the rule in. I've never had it actually conflict.

Reply:
Yeah, let's leave it in and see if that breaks anything.